PopPUNK/web.py

```diff
@@ -0,0 +1,230 @@
+import json
+import h5py
+import os 
+import re
+import sys
+import numpy as np
+import pandas as pd
+import graph_tool.topology
+import requests 
+
+def default_options():
+    """Default options for WebAPI"""
+    class args:
+        ref_db = "GPS_v3_references" 
+        output = "output"
+        q_files = os.path.join(output, "queries.txt")
+        update_db = False
+        write_references = False
+        distances = os.path.join(ref_db, ref_db + ".dists")
+        threads = 1
+        overwrite = True
+        plot_fit = 0
+        max_a_dist = 0.5
+        model_dir = ref_db
+        strand_preserved = False
+        previous_clustering = ref_db
+        external_clustering = None
+        core_only = False
+        accessory_only = False
+        assign_lineage = True
+        rank = 1
+        exact_count = False
+        web = True
+        
+        #Options required by setupDBFuncs()
+        gpu_sketch = False
+        deviceid = 0
+        gpu_dist = False
+        min_kmer_count = 0
+        min_k = 14
+        max_k = 29
+        k_step = 3
+        use_accessory = False
+        use_mash = False
+        codon_phased = False
+        microreact = False
+        cytoscape = False
+        overwrite = True
+        phandango = False
+        grapetree = False
+        info_csv = None
+        rapidnj = None
+        perplexity = 20.0
+        existing_scheme = None
+        rank_list = None
+        sketch_sizes = 10000
+        no_stream = False
+        mash = None
+
+    return (args)
+
+def get_colours(query, clusters):
+    """Colour array for Plotly.js"""
+    colours = []
+    query = '"' + str(query) + '"'
+    for clus in clusters:
+        if not clus == query:
+            colours.append('"blue"')
+        else:
+            colours.append('"rgb(255,128,128)"')
+    return colours
+
+def sketch_to_hdf5(sketch, output):
+    """Convert JSON sketch to query hdf5 database"""
+    kmers = []
+    dists = []
+
+    sketch_dict = json.loads(sketch)
+    qNames = ["query"]
+    queryDB = h5py.File(os.path.join(output, output + '.h5'), 'w')
+    sketches = queryDB.create_group("sketches")
+    sketch_props = sketches.create_group(qNames[0])
+
+    for key, value in sketch_dict.items():
+        try:
+            kmers.append(int(key))
+            dists.append(np.array(value))
+        except:
```

Comment:
I think it's probably best to specify the error type i.e. `except AttributeError as e`. You can then print e in the else block too.
Otherwise this will catch all errors, even ones you don't expect.